source/Nuke.Common/Utilities/String.IndexOf.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Copyright 2019 Maintainers of NUKE.
+// Distributed under the MIT License.
+// https://github.com/nuke-build/nuke/blob/master/LICENSE
+
+using JetBrains.Annotations;
+using System.Text.RegularExpressions;
+
+namespace Nuke.Common.Utilities
+{
+    public static partial class StringExtensions
+    {
+        [Pure]
+        public static int IndexOfRegex(this string text, string expression)
+        {
+            var regex = new Regex(expression, RegexOptions.Compiled);
+            return regex.Match(text).Index + 1;
```

Comment:
I'm curious why `index + 1` ?

Reply:
So it really depends on the regular expression, however for the one that teamcity uses it actually matches the character before the last = sign, so the +1 moves the index to the actual position of the equal sign. I think it's probably better if this method doesn't do the +1, and instead have the caller worry about such things. I will make a small update to the PR.

Reply:
That's what I thought. Thanks!